scaladoc/src/dotty/tools/scaladoc/tasty/NameNormalizer.scala

```diff
@@ -17,6 +17,14 @@ object NameNormalizer {
       val escaped = escapedName(constructorNormalizedName)
       escaped
     }
+    
+    def normalizedFullName: String = {
+      import reflect.*
+      val fullName = s.fullName
+      val withoutObjectSuffix = if s.flags.is(Flags.Module) then fullName.stripSuffix("$") else fullName
+      val escaped = escapedName(withoutObjectSuffix)
+      escaped
+    }
```

Comment:
That's close, but it will only normalize the last name in the chain. e.g. for this snippet:

```scala
object AModule {
  object AnotherModule {
    def kw: Int = 4274
  }
}
```

The full name for `AnotherModule` will be `AModule$.AnotherModule`.
Try something like this:

```suggestion
    def ownerNameChain: List[String] = {
      import reflect.*
      if s.isNoSymbol then List.empty
      else if s == defn.EmptyPackageClass then List.empty
      else if s == defn.RootPackage then List.empty
      else if s == defn.RootClass then List.empty
      else s.owner.ownerNameChain :+ s.normalizedName
    }

    def normalizedFullName: String =
      s.ownerNameChain.mkString(".")
```

Reply:
Thank you for the feedback, corrected and tested 👍